### PR TITLE
Test the weekly adherence report worker

### PR DESCRIPTION
**Pull request by alxdarksage**
Also updating the dependencies of the package to more recent integration test changes.

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            for (WeeklyAdherenceReport report : list.getItems()) {
+                if (report.getParticipant().getIdentifier().equals(userId)) {
+                    return true;
+                }
+            }
+            offset += 100;
+        }
+        return false;
+    }
+
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java:151]: This is silly, we can just search for the email address. I'll change this.

File: src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        String requestText = "{\"service\":\"WeeklyAdherenceReportWorker\", \"body\":{\"appId\":\"api\",\"studyId\":\"study1\"}}";
+        ObjectNode requestNode = (ObjectNode) DefaultObjectMapper.INSTANCE.readTree(requestText);
+
+        sqsHelper.sendMessageAsJson(workerSqsUrl, requestNode, 0);
+        
+        // Wait. Let the worker do its thing.
+        Thread.sleep(8000L);
+        
+        // This should return our user...
+        assertTrue( findUser() );
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java]: Should there be a report generated that we should be looking at?

Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java]: This was a dumb one on my part, this helper method *did* look for the existence of the report, and I changed it. I've reverted it, and renamed the method to what it actually does (it looks for the report, not the user).

File: src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java [hunk 3]
```diff
@@ -119,7 +119,8 @@ public void afterMethod() throws IOException {
 
     @Test
     public void requestCaching() throws Exception {
-        String requestText = "{\"service\":\"WeeklyAdherenceReportWorker\", \"body\":{\"appId\":\"api\",\"studyId\":\"study1\"}}";
+        String requestText = "{\"service\":\"WeeklyAdherenceReportWorker\", \"body\":{\"selectedStudies\":{\"api\":[\"study1\"]}}}";
+        // String requestText = "{\"service\":\"WeeklyAdherenceReportWorker\", \"body\":{\"appId\":\"api\",\"studyId\":\"study1\"}}";
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java]: Minor: Remove commented out code.

File: src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java [hunk 4]
```diff
@@ -128,22 +129,31 @@ public void requestCaching() throws Exception {
...
+                    .testFilter(TestFilter.TEST).pageSize(100).offsetBy(offset);
+            list = adherenceApi.getStudyParticipantWeeklyAdherenceReports(
+                    STUDY_ID_1, search).execute().body();
+            for (WeeklyAdherenceReport report : list.getItems()) {
+                if (report.getParticipant().getIdentifier().equals(userId)) {
+                    return true;
+                }
+            }
+            offset += 100;
         }
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java]: Do reports get clean up when users get deleted? If not, then this loop will take longer and longer as our list of reports grows.

As an alternative, is there a way to search adherence reports by time range and/or filter by participant?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java]: Yes, they are cascade deleted if either the study or the user is deleted.

File: src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java]: Removed

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**